qa/integration-tests/src/test/java/io/camunda/zeebe/it/gateway/GatewayAuthenticationKeycloakIT.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.it.gateway;
+
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import dasniko.testcontainers.keycloak.KeycloakContainer;
+import io.camunda.zeebe.client.CredentialsProvider;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.impl.oauth.OAuthCredentialsProviderBuilder;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import io.camunda.zeebe.qa.util.testcontainers.ZeebeTestContainerDefaults;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.zeebe.containers.cluster.ZeebeCluster;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Network;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+public class GatewayAuthenticationKeycloakIT {
+
+  public static final BpmnModelInstance PROCESS_MODEL =
+      Bpmn.createExecutableProcess("model").startEvent().endEvent().done();
+  private static final String KEYCLOAK_CONTEXT_PATH = "/auth";
+  private static final String KEYCLOAK_PATH_CAMUNDA_REALM =
+      KEYCLOAK_CONTEXT_PATH + "/realms/camunda-platform";
+  private static final String ZEEBE_CLIENT_ID = "zeebe";
+  private static final String ZEEBE_CLIENT_AUDIENCE = "zeebe-api";
+  private static final String ZEEBE_CLIENT_SECRET = "zecret";
+  private static final Network NETWORK = Network.newNetwork();
+
+  @Container
+  private static final KeycloakContainer KEYCLOAK =
+      new KeycloakContainer("quay.io/keycloak/keycloak:19.0")
+          .withContextPath(KEYCLOAK_CONTEXT_PATH)
+          .withRealmImportFile("/keycloak/camunda-platform-realm.json")
```

Comment:
complexity is in this config file, there is a risk it's not in sync with the identity setup